transip/client.go

```diff
@@ -0,0 +1,97 @@
+package transip
+
+import (
+	"context"
+	"time"
+
+	"github.com/transip/gotransip"
+	transipdomain "github.com/transip/gotransip/domain"
+	"github.com/libdns/libdns"
+)
+
+// NewSession initializes the AWS client
+func (p *Provider) NewSession() error {
```

Comment:
This is not an AWS client, is it?

I just realized that the Route53 provider has this problem, in that it seems to require provisioning. Ideally, no provisioning would be required. Is this really necessary?

If so, it needs to be clearly documented.

Btw, if this is to be used by Caddy in the caddy-dns repositories, that type will need a `Provision()` method to call this for its Caddy module. I just raised a similar point over in route53: https://github.com/caddy-dns/route53/issues/2

Reply:
I think this is just my lack of understaning how this should work. What's need is (obviously) a AccountName and a location of the PrivateKeyFile for the API to be connected. I couldn't directly figure this out how this should be done. Any pointers or some example code how this is suppoed to work in general?

Reply:
Is calling `NewClient` expensive? If not, just make this unexported and have each of your exported methods call this when needed.

I'm only giving suggestions here, you're more familiar with TransIP of course :)